docs/performance/inserts/methods.txt

```diff
@@ -0,0 +1,283 @@
+==============
+Insert Methods
+==============
+
+There are multiple ways to insert data into a CrateDB database.
+
+Some insert methods can be faster than others, depending on your setup.
+Choosing the best insert method is an easy way to improve insert performance.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+.. _insert_statement_types:
+
+Statement Types
+===============
+
+There are three types of insert statement:
+
+- :ref:`Single inserts <inserts_single_inserts>`
+- :ref:`UNNEST <inserts_unnest>`
+- :ref:`Multiple value expressions <inserts_multiple_values>`
+
+Each type of statement can be issued using any of the three types of
+:ref:`client approaches <inserts_client_approaches>`, which we cover in
+the next section.
+
+.. _inserts_single_inserts:
+
+Single Inserts
+--------------
+
+`Single inserts`_ are the most basic sort of insert statement, and look like
+this:
+
+.. code-block:: psql
+
+    cr> INSERT INTO my_table (column_a) VALUES ("value 1");
+    INSERT OK, 1 row affected  (... sec)
+
+Single inserts are typically very fast with CrateDB. A small cluster can
+easily handle several thousand inserts per second.
+
+However, it's important to note that:
+
+ 1. Every insert is first applied to the primary shard
+
+ 2. After the primary shard has been updated, the insert is then individually
+    communicated in parallel to every configured replica shard
+
+ 3. CrateDB will not return a response until all replica shards have been
+    updated
+
+The overhead for each insert (query parsing, planning, and execution, which
+includes network connection setups, connection teardowns, and connection
+metadata) starts to add up for very heavy insert workloads.
+
+In addition, lots of internal traffic will congest your network, which will
+slow down any network-based cluster operations (i.e. inserts, distributed
+reads, replication, and cluster management).
+
+.. _inserts_unnest:
+
+``UNNEST``
+----------
+
+The `UNNEST`_ function produce rows, like so:
+
+.. code-block:: psql
+
+    cr> SELECT *
+    ...   FROM UNNEST(
+    ...          [1, 2, 3],
+    ...          ['Arthur', 'Trillian', 'Marvin']);
+    +------+----------+
+    | col1 | col2     |
+    +------+----------+
+    |    1 | Arthur   |
+    |    2 | Trillian |
+    |    3 | Marvin   |
+    +------+----------+
+    SELECT 3 rows in set  (... sec)
+
+Combine ``UNNEST`` with ``INSERT`` to insert multiple rows at once:
+
+.. code-block:: psql
+
+    cr> INSERT INTO my_table (id, name)
+    ...   (SELECT *
+    ...      FROM UNNEST(
+    ...             [1, 2, 3],
+    ...             ['Arthur', 'Trillian', 'Marvin']));
+    INSERT OK, 3 rows affected  (... sec)
+
+You should see a dramatic improvement in performance over single inserts.
+
+Specifically, the advantages are:
+
+- Significantly less internal network traffic
+
+- The query only needs to be parsed, planned, and executed once
+
+- If `translog.durability`_ is set to ``REQUEST`` (the default), an insert
```

Comment:
It's once per shard that is being written to. Depending on the amount of rows being inserted and the number of shards you might end up writting to multiple shards.

Reply:
changed to:

> If `translog.durability`_ is set to ``REQUEST`` (the default), an insert
  using ``UNNEST`` flushes the disk once for every shard written to.